source/src/ModuleApplication.cc

```diff
@@ -495,6 +658,24 @@ namespace dqm4hep {
       processEvent->setPriority(ModuleApplication::PROCESS_CALL);
       m_eventLoop.postEvent(processEvent);
     }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    void ModuleApplication::openArchive(core::Run &run) {
+      if(nullptr != m_archiver) {
+        const int runNumber = m_archiverRunNumber ? run.runNumber() : -1;
+        THROW_RESULT_IF(core::STATUS_CODE_SUCCESS, !=, m_archiver->open(m_inputArchiveName, m_archiveOpenMode, m_allowOverwrite, runNumber));        
+      }
```

Comment:
Maybe still print a debug if m_archiver is null. Might be on purpose if you don't want to archive.... but it could be a problem

Reply:
I put an info message instead. Thanks !